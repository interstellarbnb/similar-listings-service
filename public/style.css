body {
    font-family: Circular,"Helvetica Neue",Helvetica,Arial,sans-serif;
    font-size: 14px;
    line-height: 1.43;
    color: #484848;
    background-color: #fff;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}
