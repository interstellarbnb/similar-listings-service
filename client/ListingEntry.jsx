import React from 'react';
import { shape, number, string, array, func } from 'prop-types';
import ReactStars from 'react-stars';
import style from './entrystyle.css';

const ListingEntry = ({ listing, openModal }) => (
  <div>
    <div className={style.heart}>
      <svg
        viewBox="0 0 32 32"
        fill="#484848"
        fillOpacity="0.5"
        stroke="#ffffff"
        strokeWidth="2.5"
        aria-label="Add listing to a list"
        role="img"
        strokeLinecap="round"
        strokeLinejoin="round"
        style={{
          height: '24px',
          width: '24px',
          display: 'block',
        }}
        onClick={() => openModal(listing)}
      >
        <path d="m23.99 2.75c-.3 0-.6.02-.9.05-1.14.13-2.29.51-3.41 1.14-1.23.68-2.41 1.62-3.69 2.94-1.28-1.32-2.46-2.25-3.69-2.94-1.12-.62-2.27-1-3.41-1.14a7.96 7.96 0 0 0 -.9-.05c-1.88 0-7.26 1.54-7.26 8.38 0 7.86 12.24 16.33 14.69 17.95a1 1 0 0 0 1.11 0c2.45-1.62 14.69-10.09 14.69-17.95 0-6.84-5.37-8.38-7.26-8.38" />
      </svg>
    </div>
    <img src={listing.imageUrl} alt="" />
    <p className={style.type}>Entire {listing.type}, {listing.bedCount} {listing.bedCount === 1 ? 'Bed' : 'Beds'}</p>
    <p className={style.title}>{listing.title}</p>
    <p className={style.small}>${listing.price} per night</p>
    <div>
      <span>
        <ReactStars
          count={5}
          size={13}
          value={listing.avgRating}
          color1="A0A0A0"
          color2="#008489"
          edit={false}
        />
      </span>
      <span className={style.review}>
        {listing.reviews.length}
      </span>
    </div>
  </div>
);

ListingEntry.propTypes = {
  listing: shape({
    id: number.isRequired,
    title: string.isRequired,
    price: number.isRequired,
    imageUrl: string.isRequired,
    reviews: array.isRequired,
    avgRating: number.isRequired,
    type: string.isRequired,
    bedCount: number.isRequired,
  }).isRequired,
  openModal: func.isRequired,
};

export default ListingEntry;
