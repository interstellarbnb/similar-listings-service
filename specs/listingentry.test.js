import React from 'react';
import { shallow, mount } from 'enzyme';
import style from '../client/entrystyle.css';

import ListingEntry from '../client/ListingEntry';

describe('<ListingEntry />', () => {
  const listing = {
    id: 9,
    title: 'Refined Black Hole',
    price: 662,
    imageUrl: 'https://source.unsplash.com/333x222/?crater',
    reviews: [3, 5, 1],
    avgRating: 3,
    type: 'Alpha Base',
    bedCount: 1,
    city: 'West Horace',
    state: 'UT',
    country: 'Lesotho',
  };
  let wrapper;
  let mockOpenModal = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<ListingEntry
      listing={listing}
      key="1"
      class="1"
      openModal={mockOpenModal}
    />);
    return wrapper;
  });

  test('Should render component', (done) => {
    expect(wrapper).toMatchSnapshot();
    done();
  });

  test('Bed should be singular when bedCount equals 1', (done) => {
    expect(wrapper.find(`.${style.type}`).text()).toContain('Bed');
    done();
    listing.bedCount = 2;
    const bedCountWrapper = shallow(<ListingEntry
      listing={listing}
      key="1"
      class="1"
      openModal={jest.fn()}
      />);
      expect(bedCountWrapper.find(`.${style.type}`).text()).toContain('Beds');
  });

  test('Should call openModal when heart is clicked', (done) => {
    wrapper.find('svg').simulate('click');
    expect(mockOpenModal.mock.calls.length).toBe(1);
    done();
  });
});
