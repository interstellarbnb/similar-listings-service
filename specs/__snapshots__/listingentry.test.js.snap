// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListingEntry /> Should render component 1`] = `
<div
  className="item"
>
  <div
    className="heart"
  >
    <svg
      aria-label="Add listing to a list"
      fill="#484848"
      fillOpacity="0.5"
      onClick={[Function]}
      role="img"
      stroke="#ffffff"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2.5"
      style={
        Object {
          "display": "block",
          "height": "24px",
          "width": "24px",
        }
      }
      viewBox="0 0 32 32"
    >
      <path
        d="m23.99 2.75c-.3 0-.6.02-.9.05-1.14.13-2.29.51-3.41 1.14-1.23.68-2.41 1.62-3.69 2.94-1.28-1.32-2.46-2.25-3.69-2.94-1.12-.62-2.27-1-3.41-1.14a7.96 7.96 0 0 0 -.9-.05c-1.88 0-7.26 1.54-7.26 8.38 0 7.86 12.24 16.33 14.69 17.95a1 1 0 0 0 1.11 0c2.45-1.62 14.69-10.09 14.69-17.95 0-6.84-5.37-8.38-7.26-8.38"
      />
    </svg>
  </div>
  <img
    alt=""
    src="https://source.unsplash.com/333x222/?crater"
  />
  <p
    className="type"
  >
    Entire 
    Alpha Base
    , 
    1
     
    Bed
  </p>
  <p
    className="title"
  >
    Refined Black Hole
  </p>
  <p
    className="small"
  >
    $
    662
     per night
  </p>
  <div>
    <span>
      <ReactStars
        char="â˜…"
        color1="A0A0A0"
        color2="#008489"
        count={5}
        edit={false}
        half={true}
        onChange={[Function]}
        size={13}
        value={3}
      />
    </span>
    <span
      className="review"
    >
      3
    </span>
  </div>
</div>
`;
